azerothcore/azerothcore-wotlk
# fix(Core/Professions):Leatherworking Specialty can't be unlearned

## Changes Proposed:
-  adds gossip option to each specialist trainer to allow you to unlearn the specialization
-  adds gossip options to the "Soothsaying for Dummies" book in Steamwheedle Port to learn any of the 3 specializations once you have learned and then unlearned one of them.

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7987

## SOURCE:
1. https://wowpedia.fandom.com/wiki/Patch_2.0.3#Professions
2. https://www.youtube.com/watch?v=yozQJdqJI3g&t=58s&ab_channel=ChillCraft
3. more sources in linked issue

## Tests Performed:
- Builds without errors
- Tested in game


## How to Test the Changes:
1. Make character at least level 49 with minimum 225 leatherworking
2. .teleport steamwheedleport and then go to hut to north and interact with book.  Verify that you do not have an option to learn one of the specializations
2. .go to one of the Specialization trainers, do their quest, and then learn specialization
Alliance:
.go c 5567
.go c 35885
.go c 51231

Horde:
.go c 11315
.go c 6931
.go c 540

3. Unlearn specialization and then try step 4 or 5
4. go to a different trainer to try learning one of the other specializations
5. .teleport steamwheedleport and then go to hut to north and interact with book.  Should give you the option to learn any of the 3 specializations.

## Known Issues and TODO List:
- [x] talking to each trainer gives an option to unlearn their specialization once learned
- [x] interacting with the book after learning and then unlearning any of the 3 gives you the option to relearn any of them.
- [x] need to add more dialogue and conditions as linked patch notes say that each of the 3 trainers should also give the option to relearn any of the 3 instead of only being able to switch specializations by using the book.
![image](https://user-images.githubusercontent.com/36058236/138541112-32e0117b-5053-4864-9ffb-ff83cb2248c7.png)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
